server/src/main/java/io/crate/expression/scalar/arithmetic/IntervalArithmeticFunctions.java

```diff
@@ -0,0 +1,132 @@
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ),
+            (signature, boundSignature) ->
+                new IntervalIntervalArithmeticScalar(Period::minus, signature, boundSignature)
+        );
+        module.register(
+                Signature.scalar(
+                ArithmeticFunctions.Names.MULTIPLY,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ),
+            MultiplyIntervalByIntegerScalar::new
+        );
+        module.register(
+                Signature.scalar(
+                ArithmeticFunctions.Names.MULTIPLY,
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ),
+            MultiplyIntervalByIntegerScalar::new
+        );
+    }
+
+    private static class IntervalIntervalArithmeticScalar extends Scalar<Period, Object> {
+
+        private final BiFunction<Period, Period, Period> operation;
+
+        IntervalIntervalArithmeticScalar(BiFunction<Period, Period, Period> operation,
+                                         Signature signature,
+                                         BoundSignature boundSignature) {
+            super(signature, boundSignature);
+            this.operation = operation;
+        }
+
+        @Override
+        public Period evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+            Period fst = (Period) args[0].value();
+            Period snd = (Period) args[1].value();
+
+            if (fst == null || snd == null) {
+                return null;
+            }
+            return operation.apply(fst, snd);
+        }
+    }
+
+    private static class MultiplyIntervalByIntegerScalar extends Scalar<Period, Object> {
+
+        private final int integerIdx;
+        private final int periodIdx;
+
+        MultiplyIntervalByIntegerScalar(Signature signature, BoundSignature boundSignature) {
+            super(signature, boundSignature);
+            var firstArgType = boundSignature.argTypes().get(0);
+            if (firstArgType.id() == IntervalType.ID) {
+                periodIdx = 0;
+                integerIdx = 1;
+            } else {
+                periodIdx = 1;
+                integerIdx = 0;
+            }
+        }
+
+        @Override
+        public Period evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+            Integer integer = (Integer) args[integerIdx].value();
+            Period period = (Period) args[periodIdx].value();
+
+            if (integer == null || period == null) {
+                return null;
+            }
+            return period.multipliedBy(integer);
```

Comment:
I've been thinking that we can bypass the overflow with something like this:
```
var millis = IntervalType.toStandardDuration(period).multiply(BigInteger.valueOf(integer)).longValueExact();
return new Period(millis).normalizedStandard(PeriodType.yearMonthDayTime());
```
but then the `normalizedStandard` doesn't do a good job to actually normalize all units and create more years/months out of days, etc.

Would you like to try out a custom normalization that we could also use in other places to create more user friendly return values for the `Period`?

Reply:
Thank you for the feedback.

I can certainly try to implement custom normalization. However, there is one aspect that concerns me a bit, which, in my opinion, is related to this topic. Currently, CrateDB returns `true` for the following query:

```sql
SELECT ('1 year'::interval) = ('365 days'::interval);
```

This behavior seems to be reasonable. However, PostgresQL, for the exact same query returns `false`. 
Apparently, in PostgresQL, one year is considered to have 360 days, because for the following query it returns `true`:

```sql
SELECT ('1 year'::interval) = ('360 days'::interval);
```

Even though, it appeared to me to be invalid, after deeper consideration it makes sense. Let’s assume that:

```
1 year = 365 days = 31536000000 milliseconds
1 month = 30 days = 2592000000 milliseconds
```

Now, if we normalize 31536000000 + 31535999999 milliseconds, the expected result is:

```
1 year 11 months 29 days 23 hours 59 minutes 59 seconds 999 milliseconds
```

However, if we follow the above assumption of 365 days in a year, we would get:

```
1 year 12 months 4 days 23 hours 59 minutes 59 seconds 999 milliseconds
```

which doesn't seem to be correct.

I'm wondering if CrateDB should adopt PostgreSQL's assumption regarding the number of days in a year. I presume that switching to 360 days in comparisons now could potentially break some client code if it relies on the fact that one year has 365 days.

What do you think about it? Am I missing something?

Reply:
Thx a lot for the taking the time to investigate!

It seems that PostgreSQL is using some more weird calculations:
```
matriv=> select extract('epoch' from '360 days'::interval);
     extract     
-----------------
 31104000.000000
(1 row)

matriv=> select extract('epoch' from '365 days'::interval);
     extract     
-----------------
 31536000.000000
(1 row)

matriv=> select extract('epoch' from '1 year'::interval);
     extract     
-----------------
 31557600.000000
(1 row)

matriv=> select extract('epoch' from '12 months'::interval);
     extract     
-----------------
 31557600.000000
(1 row)

matriv=> select extract('epoch' from '1 month'::interval);
    extract     
----------------
 2592000.000000
(1 row)

matriv=> select extract('epoch' from '30 days'::interval);
    extract     
----------------
 2592000.000000
(1 row)

matriv=> select extract('epoch' from '1 month'::interval) * 12;
    ?column?     
-----------------
 31104000.000000
(1 row)

matriv=> select extract('epoch' from '1 day'::interval) * 12 * 30;
    ?column?     
-----------------
 31104000.000000
(1 row)
```

Although 1year and 12 months yield the same result, it's not the same as 360 days nor 365 days. Checkout also that `epoch(1 month) * 12` is also different than `epoch(12 months)` :)

Regarding our `toStandardDuration` this is currently used only for comparison of intervals, so as long as the calculation is consistent for both intervals the comparison works correctly.

I'd say for the time being we can keep your original implementation with the overflow and we can open an issue to revisit this whole thing regarding normalization later on.

@piotrrzysko any more thoughts from you side?

@BaurzhanSakhariev what's your opinion?

Reply:
While researching this topic, I implemented a function that normalizes periods according to what we've discussed in this PR. Currently, the function assumes that a year has 360 days. I can share it either as part of a new PR or include it in this one.

If normalization and mitigating overflows are important for the project, I'd be happy to work on it. I can start by describing our findings in a new issue. Then, I'll probably need to discuss with some of you the direction we want to take since I'm not sure how closely we want to align with Postgres in terms of the API.

Reply:
I'd say to move forward with this PR as is and create a new issue regarding the normalization and mention that we can also "solve" the multiplication overflow.
@piotrrzysko could you open this issue please?

Reply:
> It seems that PostgreSQL is using some more weird calculations:
> 
> ```
> matriv=> select extract('epoch' from '360 days'::interval);
>      extract     

Looks like there is unreleased https://github.com/postgres/postgres/commit/f2a2bf66c87e14f07aefe23cbbe2f2d9edcd9734. 

Just noted in case we would want to go further with https://github.com/crate/crate/issues/14798 in the future and would rely on extract output.